### DISCUSSION TITLE
How to add all contents of .env as a build dependency?

### DISCUSSION THREAD

**Discussion by kevin-krug** [vercel/turborepo#687] [Help] [answered]
As per docs it's possible to define the contents of `.env` as a global dependency, https://turborepo.org/docs/reference/configuration#globaldependencies

However if I were to add `.env` as a workspace build dependency turbo will think its a task to run. So far I would need to list each env variable separately. 

doesn't work:
```
{
  "pipeline": {
    ...
    "web#build": {
      "dependsOn": [
        "^build",
        ".env"
      ],
      "outputs": [".next/**"]
    }
  }
}
```
works:
```
{
  "pipeline": {
    ...
    "web#build": {
      "dependsOn": [
        "^build",
        "$SECRET_KEY"
      ],
      "outputs": [".next/**"]
    }
  }
}
```

Is there a way currently to add all contents of `.env` as a workspace build dependency?

**Comment by averageokguy** [comment]
@kevin-krug not sure if you're still working through this (or if this answers your question), but i think env vars go in "dependsOn": [...] and files/globs go in "inputs": [...]. you should be able to add ".env" to inputs arr instead of depends on arr successfully => should consider contents of .env file as a whole in hash fn, so if even one k/v changes, the overall hash changes
also, hello other kevin

**Comment by anthonyshew** [comment, marked as answer]
This is solved by [the `globalDotEnv` property](https://turbo.build/repo/docs/reference/configuration#globaldotenv) in today.
